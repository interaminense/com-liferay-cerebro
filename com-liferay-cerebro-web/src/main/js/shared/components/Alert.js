import autobind from 'autobind-decorator';
import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';

import Icon from './Icon';

const TYPES = [
	'danger',
	'dark',
	'info',
	'light',
	'primary',
	'secondary',
	'success',
	'warning'
];

class AlertLink extends Component {
	render() {
		return (
			<a {...this.otherProps()} class="alert-link">
				{this.props.children}
			</a>
		);
	}
}

class Alert extends Component {
	@autobind
	handleClose() {
		const {id, onClose} = this.props;

		onClose(id);
	}

	render() {
		const {onClose, children, notification, title, type} = this.props;

		const classes = getCN('alert', {
			'alert-dismissible': onClose,
			[`alert-${type}`]: type,
			'alert-notification': notification
		});

		let content = children;

		if (notification) {
			content = <p>{children}</p>;
		}

		return (
			<div class={classes} role="alert">
				{onClose && (
					<button class="close" onClick={this.handleClose}>
						<Icon symbol="times" />
					</button>
				)}

				{title && <strong class="lead">{title}</strong>}

				{content}
			</div>
		);
	}
}

Alert.PROPS = {
	notification: Config.bool().value(false),
	onClose: Config.func(),
	title: Config.string(),
	type: Config.oneOf(TYPES)
};

Alert.AlertLink = AlertLink;
Alert.TYPES = TYPES;

export default Alert;