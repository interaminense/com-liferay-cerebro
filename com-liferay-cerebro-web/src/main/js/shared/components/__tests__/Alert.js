import Alert from '../Alert';

describe('Alert', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new Alert();

		expect(component).toMatchSnapshot();
	});

	it('should render as a notification alert', () => {
		component = new Alert({
			notification: true
		});

		expect(component).toMatchSnapshot();
	});

	it('should render as dismissable', () => {
		component = new Alert({
			onClose: () => {}
		});

		expect(component).toMatchSnapshot();
	});

	it('should render with a title', () => {
		component = new Alert({
			title: 'hello world'
		});

		expect(component).toMatchSnapshot();
	});

	it('should render with a specific type', () => {
		component = new Alert({
			type: Alert.TYPES[1]
		});

		expect(component).toMatchSnapshot();
	});

	it('should render with content', () => {
		component = new Alert({
			children: ['hello children']
		});

		expect(component).toMatchSnapshot();
	});

	it('should render with multiple config set', () => {
		component = new Alert({
			children: ['hello children'],
			notification: true,
			onClose: () => {},
			title: 'this is a title'
		});

		expect(component).toMatchSnapshot();
	});
});