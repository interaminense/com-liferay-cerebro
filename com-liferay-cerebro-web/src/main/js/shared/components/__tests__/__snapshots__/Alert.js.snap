// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render 1`] = `<div class="alert" role="alert"></div>`;

exports[`Alert should render as a notification alert 1`] = `
<div class="alert alert-notification" role="alert">
  <p></p>
</div>
`;

exports[`Alert should render as dismissable 1`] = `
<div class="alert alert-dismissible" role="alert">
  <button class="close">
    <svg class="icon-root lexicon-icon lexicon-icon-times">
      <use xlink:href="/images/lexicon/icons.svg#times"></use>
    </svg>
  </button>
</div>
`;

exports[`Alert should render with a specific type 1`] = `<div class="alert alert-dark" role="alert"></div>`;

exports[`Alert should render with a title 1`] = `
<div class="alert" role="alert">
  <strong class="lead">hello world</strong>
</div>
`;

exports[`Alert should render with content 1`] = `<div class="alert" role="alert">hello children</div>`;

exports[`Alert should render with multiple config set 1`] = `
<div class="alert alert-dismissible alert-notification" role="alert">
  <button class="close">
    <svg class="icon-root lexicon-icon lexicon-icon-times">
      <use xlink:href="/images/lexicon/icons.svg#times"></use>
    </svg>
  </button>
  <strong class="lead">this is a title</strong>
  <p>hello children</p>
</div>
`;
